protocol/x/clob/keeper/msg_server_place_order.go

```diff
@@ -115,31 +116,35 @@ func (k Keeper) HandleMsgPlaceOrder(
 
 	// 4. Emit the new order placement indexer event.
 	if order.IsConditionalOrder() {
-		k.GetIndexerEventManager().AddTxnEvent(
-			ctx,
-			indexerevents.SubtypeStatefulOrder,
-			indexerevents.StatefulOrderEventVersion,
-			indexer_manager.GetBytes(
-				indexerevents.NewConditionalOrderPlacementEvent(
-					order,
+		if !isInternalOrder {
```

Comment:
is this logic right? we never emit any order placements for vault orders?

Reply:
by emitting a replacement event (instead of placement), does indexer / front-end behavior change in terms of storing / displaying vault orders?

Reply:
@dydxwill I think it would make sense here to check if `ordersToCancel[i]` is null (the first time vault places orders, nothing to cancel yet), then send an order place message instead of an order replacement

@tqin7 The goal here is to not have it change other than remove the flickering

Reply:
@tqin7 would ordersToCancel[i] ever be nil here or will `k.GetVaultClobOrders(
		ctx.WithBlockHeight(ctx.BlockHeight()-1),
		vaultId,
	)` always return something even if it's the first block?

Reply:
`ordersToCancel` can be empty (for example if vault equity is <= 0 or clob pair doesn't exist). Good to check whether `ordersToCancel` is empty